EphraimDev/join-u
# Added user signup route with validations

#### What does this PR do?
Add the controller and route for user sign up authentication
#### Description of Task to be completed?
Have the following endpoint working
POST /api/v1/auth/signup
#### How should this be manually tested?
After cloning the repo, cd into it, set the environment variables and run `npm run start:dev`
* Using postman, test for the above enpoint
* The body of the request should contain at least `first_name` `last_name` `email` `password`
#### What are the relevant pivotal tracker stories?
#167689371
